UnitTests/BraintreeTestShared/MockAPIClient.swift

```diff
@@ -92,6 +93,10 @@ public class MockAPIClient: BTAPIClient {
         postedPayPalContextID = payPalContextID
         postedLinkType = linkType
         postedAnalyticsEvents.append(name)
+
+        if postedShouldSendAPIRequestLatency {
+            postedAnalyticsEvents.append(BTCoreAnalytics.apiRequestLatency)
+        }
```

Comment:
Because of this, our assertions added in `BTPayPalClient_Tests` are only testing this Mock behavior, and not executing any of our new code added in BTAPIClient & BTHTTP.

We _could_ override `fetchAPITiming()` in this Mock class instead, but it'll be a similar situation of only passing based on added Mock code.

I would update BTAPIClient & BHTTP tests instead, and then the best we can do at the PayPalClient level is assert that `shouldSendAPIRequestLatency` is true.

Reply:
I updated `BTPayPalClient_Tests` to just test setting `shouldSendAPIRequestLatency` here: https://github.com/braintree/braintree_ios/pull/1292/commits/8af3d0b4452f740f97b6feb73d4fa9ad549defba. For `BTAPIClient_Tests` we are going to run into the same issue, currently those tests when testing analytics ([example here](https://github.com/braintree/braintree_ios/blob/2120f0b705d4fd1f6173b431d71a7d9705a7d096/UnitTests/BraintreeCoreTests/BTAPIClient_Tests.swift#L406-L414)) use `FakeAnalyticsService` ([here](https://github.com/braintree/braintree_ios/blob/main/UnitTests/BraintreeCoreTests/Analytics/FakeAnalyticsService.swift)) which just sends whatever event you tell it to send. So we would end up just testing our mock there as well.

Reply:
But we can assert that `FakeAnalyticsService.sendAnalyticsEvent()` was called with the expected values, to test the logic that lives within `BTAPIClient.fetchAPITiming()`.

Rough outline like:
* when networkTimingDelegate.fetchAPITiming() delegate method invoked, sends analytic event
* when delegate method invoked, and analytics path, does not send analytic event
* when delegate method invoked, and shouldSendAPILatency = false, does not send analytic event
* when delegate method invoked, and path contains merchantID, strips merchantID from posted JSON body tag param

The easiest way would be to call `sut.fetchAPITiming()` directly in each test. The probably "right" way would be to set a MockTimingDelegate on sut and trigger the delegate call that way via the mock. Not sure it's worth the setup for the later option at this point though.

Reply:
> But we can assert that FakeAnalyticsService.sendAnalyticsEvent() was called with the expected values, to test the logic that lives within BTAPIClient.fetchAPITiming().

This method does not behave at least how I would expect, because my initial thought was the same. If we take a look at the test `testSendAnalyticsEvent_whenCalled_callsAnalyticsService` ([here](https://github.com/braintree/braintree_ios/blob/2120f0b705d4fd1f6173b431d71a7d9705a7d096/UnitTests/BraintreeCoreTests/BTAPIClient_Tests.swift#L406-L414)) it is calling `FakeAnalyticsService.sendAnalyticsEvent()` with `blahblah` and asserting that the `mockAnalyticsService.lastEvent` equals `blahblah`. 

Which will always be true because that's the input we are giving it. While it is an override we are not actually hitting the override unless we call it directly so our call to `BTAPIClient.fetchAPITiming()` never actually hits `FakeAnalyticsService.sendAnalyticsEvent()`. We can take a look at this a bit later if that doesn't make sense.

Reply:
Paired on this because I was overthinking it, updated here: https://github.com/braintree/braintree_ios/pull/1292/commits/ff199133838acd50aa135b63a273fa2e31103b2c